lib/checkleakautovar.cpp

```diff
@@ -363,6 +363,54 @@ bool CheckLeakAutoVar::checkScope(const Token * const startToken,
         while (Token::Match(ftok, "%name% :: %name%"))
             ftok = ftok->tokAt(2);
 
+        // memcpy / memmove
+        if (Token::Match(varTok, "memcpy|memmove")) {
```

Comment:
I would prefer that we solve this ticket without hardcoding. I rather rely on the configuration in std.cfg if possible. If the configuration does not satisfy your requirements can you either suggest how the configuration can be improved or add some kind of bailout.

there are many other functions that behave like memcpy/memmove for instance windows api has CopyMemory. gtk could have some other function. Qt has some qmemcpy. various OSes implement such features.. So your fix will only work for a very limited number of cases.

Reply:
Maybe adding attributes in the .cfg files that indicate that a resource is moved/copied from one parameter to another? I can think of ways of guessing depending on the direction attrbute, but i think it might be to general.